synapse/cores/common.py

```diff
@@ -1419,6 +969,28 @@ def _bumpTufoCache(self, tufo, prop, oldv, newv):
                 if atlim:
                     self._delCacheKey(ckey)
 
+    def _onSetAuthEn(self, conf):
+        if not conf:
+            return
+
+        dirn = self.getConfOpt('dir')
+        if dirn is None:
+            raise s_exc.ReqConfOpt(name='dir', mesg='auth:en=1 requires a cortex dir')
+
+        path = self.genCorePath('auth')
+
+        self.auth = s_auth.Auth(path)
+        self.onfini(self.auth.fini)
+
+        name = self.getConfOpt('auth:admin')
```

Comment:
So the `auth:admin` opt is the name of admin user and there can only be 1 admin user. If this was set to `root@localhost`, name would be `root@localhost`, then we would check `self.auth.users` for `root@localhost`. If the user isn't there, we will add it. In either case, we give it admin perms.

Reply:
No-op